packages/core/src/components/accordion/components/accordion-section.tsx

```diff
@@ -0,0 +1,201 @@
+import { h, Component, ComponentInterface, Host, Element, State, Prop, FunctionalComponent, Fragment, Method } from '@stencil/core';
+import { AccordionHeading, AccordionInterface, AccordionInternalState, AccordionSectionState } from '../accordion.interfaces';
+
+@Component({
+  tag: 'dso-accordion-section',
+  styleUrl: 'accordion-section.scss',
+  shadow: true,
+})
+export class AccordionSection implements ComponentInterface {
+  private accordion?: AccordionInterface;
+
+  @Element()
+  host!: HTMLElement;
+
+  @State()
+  accordionState?: AccordionInternalState;
```

Comment:
Volgens mij is dit niet hoe die Stencil store gebruikt moet worden. Je kiest één van de twee: `@State()` of `createStore()`.

Een store wordt buiten 'n component gemaakt (`export const store = createStore();`) en kan met `import { store } from '...';` overal worden gebruikt. Wijzigingen aan de store worden in alle component getoond, zonder `forceUpdate()` of communicatie tussen componenten.

Ik begrijp waarom je 'm IN het component maakt, zodat er meerdere accordion's bestaan. Maar dan moet volgens mij die `@State()` eraf? En anders overwegen om 1 store voor meerdere componenten te maken?

Ik begrijp je conflict overigens hoor..

Reply:
In `accordion.tsx` is er geen `@State()`. Daar regelt de stencil store dat er wordt gerendered. De `accordion-section` heeft de internal state van zijn vader nodig. Die haalt hij op via een `@Method()`, deze is `async`. De `section` heeft eenmalig een trigger nodig om te renderen na het zetten van de internalState. `@State()` was hiervoor eigenlijk misbruikt. Ik los het nu op door 1-malig `forceUpdate` te doen.

`componentWillLoad` kan helaas niet async gemaakt worden, dan gaat de boel stuk.

Reply:
>A promise can be returned, that can be used to wait for the first render().

https://stenciljs.com/docs/component-lifecycle#componentwillload

Ik ben alweer vergeten wat we hiermee wilden doen 🤷

Reply:
Ja een promise returnen werkt wel zie ik nu, ik had wel issues met het `async` maken van de functie.

Reply:
Ohja!